ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/discovery/TimelineMetricMetadataManager.java

```diff
@@ -87,6 +86,8 @@
   AtomicBoolean SYNC_HOSTED_APPS_METADATA = new AtomicBoolean(false);
   AtomicBoolean SYNC_HOSTED_INSTANCES_METADATA = new AtomicBoolean(false);
 
+  private Map<String,Set<String>> appInstanceMap = new ConcurrentHashMap<>();
```

Comment:
Instead of this data structure why don't we do a Phoenix lookup with wildcard pass down?